pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -336,9 +339,25 @@ func (ev *Evaluator) prepareCandidate(ctx context.Context, c Candidate, pod *v1.
 		// Otherwise we should delete the victim.
 		if waitingPod := fh.GetWaitingPod(victim.UID); waitingPod != nil {
 			waitingPod.Reject(pluginName, "preempted")
-		} else if err := util.DeletePod(ctx, cs, victim); err != nil {
-			klog.ErrorS(err, "Preempting pod", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
-			return framework.AsStatus(err)
+		} else {
+			if feature.DefaultFeatureGate.Enabled(features.JobBackoffPolicy) {
+				condition := &v1.PodCondition{
+					Type:    v1.PreemptedByScheduler,
+					Status:  v1.ConditionTrue,
+					Message: "Preempted to accommodate a Pod with higher priority",
+				}
+				newStatus := pod.Status.DeepCopy()
+				if apipod.UpdatePodCondition(newStatus, condition) {
+					if err := util.PatchPodStatus(ctx, cs, victim, newStatus); err != nil {
+						klog.ErrorS(err, "Preparing pod preemption", "pod", klog.KObj(victim), "preemptor", klog.KObj(pod))
+						return framework.AsStatus(err)
+					}
+				}
+			}
+			if err := util.DeletePod(ctx, cs, victim); err != nil {
```

Comment:
One concern specific to the scheduler (as opposed to the other controllers). If this Delete fails (for example, because the apiserver is suddenly unavailable, or the scheduler died), it's possible that the scheduler decides to preempt a different pod next time (or none). Then, this pod would be left with a condition that the pod was preempted, when it actually wasn't.

The solution would be to have a small controller in kube-controller-manager that either completes a preemption (does a delete) or removes the condition (or set it to false) if the condition timestamp is old enough and there is no deletion timestamp.

I would not block on this for alpha, but we probably need it for beta graduation.

Reply:
Makes sense, I would vote for updating the status of the condition as false. 
For now I have added a note about this concern and updated the graduation criteria in the KEP update: https://github.com/kubernetes/enhancements/pull/3438

Reply:
This is being covered in this PR: https://github.com/kubernetes/kubernetes/pull/111475